src/collider/collider/logmanager.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+// Package collider implements a signaling server based on WebSocket.
+package collider
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+// LogManager redirects log output to files, one for each |cycle|, and deletes files that are last modified before the retention period.
+type LogManager struct {
+	dir       string
+	cycle     time.Duration
+	retention time.Duration
+	file      *os.File
+	stopped   chan bool
+	ticker    *time.Ticker
+}
+
+func NewLogManager(d string, c time.Duration, r time.Duration) *LogManager {
+	if d == "" {
+		log.Printf("Invalid log dir")
+		return nil
+	}
+	if c >= r {
+		log.Printf("Log retention time must be greater than log cycle")
+		return nil
+	}
+	return &LogManager{
```

Comment:
Can you run gofmt if you haven't already? Probably the best way to deal with style in the absence of someone with readability. Thanks.